pkg/controller/prpqr_reconciler/prpqr_reconciler.go

```diff
@@ -191,44 +211,98 @@ func (r *reconciler) reconcile(ctx context.Context, req reconcile.Request, logge
+
+			submitted := generateJobNameToSubmit(baseMetadata, inject, &prpqr.Spec.PullRequest)
+			aggregatorJob, err := generateAggregatorJob(uid, mimickedJob, jobSpec.JobName(jobconfig.PeriodicPrefix), req.Name, req.Namespace, r.prowConfigGetter.Config(), time.Now(), submitted)
+			if err != nil {
+				logger.WithError(err).Error("Failed to generate an aggregator prowjob")
+				statuses[mimickedJob] = &v1.PullRequestPayloadJobStatus{
+					ReleaseJobName: mimickedJob,
+					Status: prowv1.ProwJobStatus{
+						State:       prowv1.ErrorState,
+						Description: fmt.Errorf("failed to create an aggregator prowjob: %w", err).Error(),
+					},
+				}
+				continue
+			}
 			statuses[mimickedJob] = &v1.PullRequestPayloadJobStatus{
 				ReleaseJobName: mimickedJob,
-				Status: prowv1.ProwJobStatus{
-					State:       prowv1.ErrorState,
-					Description: fmt.Errorf("failed to create prowjob: %w", err).Error(),
-				},
+				ProwJob:        aggregatorJob.Name,
+				Status:         aggregatorJob.Status,
 			}
-			continue
+			prowjobsToCreate = append(prowjobsToCreate, aggregatorJob)
+
+		} else {
+			prowjob, err := generateProwjob(ciopConfig, r.prowConfigGetter.Config(), baseMetadata, req.Name, req.Namespace, &prpqr.Spec.PullRequest, mimickedJob, inject, nil)
+			if err != nil {
+				logger.WithError(err).Error("Failed to generate prowjob")
+				statuses[mimickedJob] = &v1.PullRequestPayloadJobStatus{
+					ReleaseJobName: mimickedJob,
+					Status: prowv1.ProwJobStatus{
+						State:       prowv1.ErrorState,
+						Description: fmt.Errorf("failed to generate prowjob: %w", err).Error(),
+					},
+				}
+				continue
+			}
+			prowjobsToCreate = append(prowjobsToCreate, prowjob)
 		}
 
-		// There is some delay until it gets back to our cache, so block until we can retrieve
-		// it successfully.
-		key := ctrlruntimeclient.ObjectKey{Namespace: prowjob.Namespace, Name: prowjob.Name}
-		if err := wait.Poll(100*time.Millisecond, 5*time.Second, func() (bool, error) {
-			if err := r.client.Get(ctx, key, &prowv1.ProwJob{}); err != nil {
-				if kerrors.IsNotFound(err) {
-					return false, nil
+		for _, prowjob := range prowjobsToCreate {
+			logger.WithField("job", prowjob.Spec.Job).Info("Creating prowjob...")
+			if err := r.client.Create(ctx, prowjob); err != nil {
+				statuses[mimickedJob] = &v1.PullRequestPayloadJobStatus{
+					ReleaseJobName: mimickedJob,
+					Status: prowv1.ProwJobStatus{
+						State:       prowv1.ErrorState,
+						Description: fmt.Errorf("failed to create prowjob: %w", err).Error(),
+					},
 				}
-				return false, fmt.Errorf("getting prowJob failed: %w", err)
+				continue
 			}
-			return true, nil
-		}); err != nil {
+
+			// There is some delay until it gets back to our cache, so block until we can retrieve
+			// it successfully.
+			key := ctrlruntimeclient.ObjectKey{Namespace: prowjob.Namespace, Name: prowjob.Name}
+			if err := wait.Poll(100*time.Millisecond, 5*time.Second, func() (bool, error) {
```

Comment:
I do not like "some delay" without actual backing the rationale (eg why 5 secs not 10, or not 15). On the other hand, one can spot this on several occasions coming across test-infra and ci-tools repos. So, it's just polemics.